polkadot/node/core/provisioner/src/disputes/prioritized_selection/mod.rs

```diff
@@ -157,10 +157,9 @@ where
 	// Filter out unconfirmed disputes. However if the dispute is already onchain - don't skip it.
 	// In this case we'd better push as much fresh votes as possible to bring it to conclusion
 	// faster.
-	let recent_disputes = recent_disputes
-		.into_iter()
-		.filter(|d| d.2.is_confirmed_concluded() || onchain.contains_key(&(d.0, d.1)))
-		.collect::<Vec<_>>();
+	recent_disputes.retain(|key, dispute_status| {
```

Comment:
nit: I'd put the retain at the declaration on 146 to avoid the `mut`.

Reply:
retain consumes self, so I guess if we want to avoid `mut` I cand do an `.into_iter().filter()` like
```rust
let mut recent_disputes = request_disputes(sender).await.into_iter().filter(|(key, dispute_status)| {
		dispute_status.is_confirmed_concluded() || onchain.contains_key(key)
	}).collect::<BTreeMap<_,_>>();
```